src/Converter.cc

```diff
@@ -229,13 +240,270 @@ void Converter::ConvertImpl(tinyxml2::XMLElement *_elem,
     {
       Remove(_elem, childElem);
     }
+    else if (name == "unnest")
+    {
+      Unnest(_elem);
+      // TODO(jenn) delete debug statements
+      // std::cout << "\n\n_elem after unnest:\n"
+      //             << PrintElement(_elem) << std::endl;
+    }
     else if (name != "convert")
     {
       sdferr << "Unknown convert element[" << name << "]\n";
     }
   }
 }
 
+/////////////////////////////////////////////////
+void Converter::Unnest(tinyxml2::XMLElement *_elem)
+{
+  SDF_ASSERT(_elem != NULL, "SDF element is NULL");
+
+  tinyxml2::XMLDocument *doc = _elem->GetDocument();
+  tinyxml2::XMLNode *copy = DeepClone(doc, _elem);
+
+  // std::cout << "copy:\n" << PrintElement(copy->ToElement()) << std::endl;
+
+  tinyxml2::XMLElement *elem = _elem->FirstChildElement(), *nextElem = nullptr;
+  while (elem)
+  {
+    nextElem = elem->NextSiblingElement();
+
+    std::string elemName = elem->Name();
+
+    // skip element if not one of the following
+    if (elemName != "frame" && elemName != "joint"
+        && elemName != "link" && elemName != "model")
+    {
+      elem = nextElem;
+      continue;
+    }
+
+    std::string attrName = elem->Attribute("name");
```

Comment:
Need to check if the attribute exists first.

Reply:
I didn't check if the attribute exists since I'm assuming the element is either a `frame`, `joint`, `link`, or `model` which require a name. Should I implement the check anyway?

Reply:
Yes, because at this stage of the parser, not much error checking has happened, so you might get an element without the name attribute. I believe the test `UNIT_ign_TEST` is failing due to this.

Reply:
I see, thanks! 1435eab8f21c88346889a1c8ac05fe62aedb6990